pkg/handler/eval_cache.go

```diff
@@ -110,12 +101,11 @@ func (ec *EvalCache) reloadMapCache() error {
 			return nil, err
 		}
 
-		ec.mapCacheLock.Lock()
-		defer ec.mapCacheLock.Unlock()
-
-		ec.idCache = idCache
-		ec.keyCache = keyCache
-		ec.tagCache = tagCache
+		ec.cache.Store(&cacheContainer{
```

Comment:
iiuc this is called from Start(). Do you expect Start() do be called concurrently? If not, why is the atomic needed here?

Reply:
Start() will call it once, however, there's a cron goroutinue also periodically updates the cache when the cache is heavily being read, so atomic is used here. Previously this is locked by RWMutex.

Reply:
does it update the cache's values or replaces the entire object?

Reply:
replaces the whole object

Reply:
cool